cli/pack/systemd_dir.go

```diff
@@ -45,10 +55,103 @@ var (
 	}
 )
 
+type SystemdUnitParams struct {
+	FdLimit           *int `yaml:"fd-limit"`
+	StateboardFdLimit *int `yaml:"stateboard-fd-limit"`
+}
+
+func parseSystemdUnitParamsFile(systemdUnitParamsPath string, defaultUnitParamsPath string) (*SystemdUnitParams, error) {
+	var fileContentBytes []byte
+	var err error
+
+	if systemdUnitParamsPath == "" {
+		if _, err := os.Stat(defaultUnitParamsPath); err == nil {
+			log.Debugf("Default file with system unit params is used: %s", systemdUnitParamsPath)
+			systemdUnitParamsPath = defaultUnitParamsPath
+		} else if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("Failed to use default file with system unit params: %s", err)
+		}
+	}
+
+	if systemdUnitParamsPath == "" {
+		return &SystemdUnitParams{}, nil
+	}
+
+	if _, err := os.Stat(systemdUnitParamsPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("Specified file with system unit params %s doesn't exists", systemdUnitParamsPath)
+	} else if err != nil {
+		return nil, fmt.Errorf("Impossible to use specified file %s: %s", systemdUnitParamsPath, err)
+	}
+
+	fileContentBytes, err = common.GetFileContentBytes(systemdUnitParamsPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read file with system unit params:  %s", err)
+	}
+
+	var systemdUnitParams SystemdUnitParams
+	if err := yaml.Unmarshal([]byte(fileContentBytes), &systemdUnitParams); err != nil {
+		return nil, fmt.Errorf("Failed to parse system unit params file %s: %s", systemdUnitParamsPath, err)
+	}
+
+	return &systemdUnitParams, nil
+}
+
+func checkAndSetDefaults(fdLimit **int, defaultValue int, isValid func() error) error {
+	if fdLimit == nil {
+		return project.InternalError("Failed to use fd limit parameter")
+	}
+
+	if *fdLimit == nil {
+		*fdLimit = new(int)
+		**fdLimit = defaultValue
+		return nil
+	}
+
+	return isValid()
+}
+
+func getSystemdUnitParams(ctx *context.Ctx) (*SystemdUnitParams, error) {
+	systemdUnitParams, err := parseSystemdUnitParamsFile(
```

Comment:
You use `:=` here, but lower `=` is used. It makes these parts dependent on each other. Let's declare `var err error` at the function beginning.